### PR TITLE
Fix #2545 - Warning about loss of changes appears multiple times

### LINKED ISSUE #2545: Inline editing message displays multiple times
#### Issue

Sometimes during inline editing message "You have clicked away..." displays two or three time in row when I click on cancel button. 
#### Expected Behavior

Message should display only once.
#### Actual Behavior

Message about inline editing displays two or three times in row.
#### Steps to Reproduce

For example when I'm editing address in Accounts module message displays two times.

#### Your Environment

* SuiteCRM 7.7.6
* Google Chrome 54.0.2840.71
* MySQL, PHP 5.6
* Debian jessie

### PR DESCRIPTION
**Pull request by kdjmonaghan**
## Description
Fixing issue #2545 - Often times the warning loss of changes dialogue window would appear when entering an inline edit box (upon double click). The Window would also be prompted if there were no changes made.

## Motivation and Context
A potential frustrating (from user interface point of view, user experience) bug that shouldn't have been present. Removal of the dialogue window appearing upon entering (sometimes) and clicking away without changes to the window should result in some less frustrating in-line editing experience.

## How To Test This
In issue #2545 the user provided a note for reproduction:
"When i mark some field for inline editing and click ouside the message show
up. I click cancel on message and second message popup immediatlety. When I
click cancel second time focus goes back to edited field."

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/InlineEditing/inlineEditing.js [hunk 1]
```diff
@@ -267,24 +267,37 @@ $(document).on('click', function (e) {
...
+                var replacementUserValue = $("#" + fieldName).val();
+
+                // Parsing empty text returns undefined, if the string returns anything other than undefined, replace
+                // user_value with this value.
+                if (replacementUserValue != undefined) {
+                    user_value = replacementUserValue;
+                }
             }
 
+            // Debug Output
```
Comment (kdjmonaghan) [inline include/InlineEditing/inlineEditing.js]: Apologies - forgot I added these to find the issue. :+1:

File: include/InlineEditing/inlineEditing.js [hunk 2]
```diff
@@ -267,24 +267,37 @@ $(document).on('click', function (e) {
             var user_value = getInputValue(field, type);
 
             /**
```
Comment (mattlorimer) [inline include/InlineEditing/inlineEditing.js]: I would prefer to put comments/ detail info on fixes into the detail part of commit message, http://chris.beams.io/posts/git-commit/#seven-rules

File: include/InlineEditing/inlineEditing.js [hunk 3]
```diff
@@ -252,20 +251,55 @@ function clickedawayclose(field,id,module, type){
...
+             * "LoadFieldHTMLValue" called on line #270, declared at line #509, comparison has to be done this way.
+             *
+             * Additionally, auto populated fields (QS fields) have '_display' at the end of their field names which
+             * throws off any form of comparison check. An additional check has to be done to ensure we're considering
+             * these field names. An alternative solution would be to ensure these names do not have a trailing
+             * '_display' in the field names in the future.
+             */
+
+            // Return user value to empty string for comparison if undefined at this stage (empty field check fix)
+            if (user_value == undefined) {
```
Comment (mattlorimer) [inline include/InlineEditing/inlineEditing.js]: if (typeof user_value === "undefined") 
more robust

File: include/InlineEditing/inlineEditing.js [hunk 4]
```diff
@@ -252,20 +251,55 @@ function clickedawayclose(field,id,module, type){
...
+                var replacementUserValue = $("#" + fieldName).val();
+
+                // Parsing empty text returns undefined, if the string returns anything other than undefined, replace
+                // user_value with this value.
+                if (replacementUserValue != undefined) {
+                    user_value = replacementUserValue;
+                }
+            }
+
+            if (user_value == outputValueParse || user_value == output_value) {
```
Comment (mattlorimer) [inline include/InlineEditing/inlineEditing.js]: just define var alertFlag = true; remove else

**Pull request merged**